### PR TITLE
Add network field to target instance.

### PR DESCRIPTION
**Pull request by upodroid**
Fixes: https://github.com/hashicorp/terraform-provider-google/issues/6985

Google is not returning the network field yet which breaks the test.

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/master/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
compute: added network field to `compute_target_instance`
```

### CODE REVIEW

**Comment by upodroid** [comment]
@danawillow second time in a week where the Google APIs are not returning/setting fields documented in the API.

```
REDACTED  MCW0CDP3YY  ~  go  …  github.com  terraform-providers  terraform-provider-google-beta   master  3✎  USAGE  $    make testacc TEST=./google-beta TESTARGS='-run=TestAccComputeTargetInstance_targetInstanceCustomNetworkExample'
==> Checking source code against gofmt...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 TF_SCHEMA_PANIC_ON_ERROR=1 go test ./google-beta -v -run=TestAccComputeTargetInstance_targetInstanceCustomNetworkExample -timeout 240m -ldflags="-X=github.com/hashicorp/terraform-provider-google-beta/version.ProviderVersion=acc"
=== RUN   TestAccComputeTargetInstance_targetInstanceCustomNetworkExample
=== PAUSE TestAccComputeTargetInstance_targetInstanceCustomNetworkExample
=== CONT  TestAccComputeTargetInstance_targetInstanceCustomNetworkExample
    TestAccComputeTargetInstance_targetInstanceCustomNetworkExample: testing.go:674: Step 0 error: After applying this step, the plan was not empty:
        
        DIFF:
        
        DESTROY/CREATE: google_compute_target_instance.custom_network
          creation_timestamp: "2020-08-10T05:35:16.111-07:00" => "<computed>"
          description:        "" => ""
          id:                 "projects/REDACTED/zones/us-central1-a/targetInstances/tf-test-custom-network4kgzu99ytj" => "<computed>"
          instance:           "https://www.googleapis.com/compute/v1/projects/REDACTED/zones/us-central1-a/instances/tf-test-cusom-network-target-vm4kgzu99ytj" => "projects/REDACTED/zones/us-central1-a/instances/tf-test-cusom-network-target-vm4kgzu99ytj"
          name:               "tf-test-custom-network4kgzu99ytj" => "tf-test-custom-network4kgzu99ytj"
          nat_policy:         "NO_NAT" => "NO_NAT"
          network:            "" => "projects/REDACTED/global/networks/default" (forces new resource)
          project:            "REDACTED" => "<computed>"
          self_link:          "https://www.googleapis.com/compute/v1/projects/REDACTED/zones/us-central1-a/targetInstances/tf-test-custom-network4kgzu99ytj" => "<computed>"
          zone:               "https://www.googleapis.com/compute/v1/projects/REDACTED/zones/us-central1-a" => "<computed>"
        
        
        
        STATE:
        
        data.google_compute_image.vmimage:
          ID = projects/debian-cloud/global/images/debian-10-buster-v20200805
          provider = provider.google
          archive_size_bytes = 9652082688
          creation_timestamp = 2020-08-05T10:50:22.077-07:00
          description = Debian, Debian GNU/Linux, 10 (buster), amd64 built on 20200805, supports Shielded VM features
          disk_size_gb = 10
          family = debian-10
          image_encryption_key_sha256 = 
          image_id = 2413963976930447282
          label_fingerprint = 42WmSpB8rSM=
          licenses.# = 1
          licenses.0 = https://www.googleapis.com/compute/beta/projects/debian-cloud/global/licenses/debian-10-buster
          name = debian-10-buster-v20200805
          project = debian-cloud
          self_link = https://www.googleapis.com/compute/beta/projects/debian-cloud/global/images/debian-10-buster-v20200805
          source_disk = 
          source_disk_encryption_key_sha256 = 
          source_disk_id = 
          source_image_id = 
          status = READY
        data.google_compute_network.target-vm:
          ID = projects/REDACTED/global/networks/default
          provider = provider.google
          description = 
          gateway_ipv4 = 
          name = default
          self_link = https://www.googleapis.com/compute/beta/projects/REDACTED/global/networks/default
          subnetworks_self_links.# = 24
          subnetworks_self_links.0 = https://www.googleapis.com/compute/beta/projects/REDACTED/regions/northamerica-northeast1/subnetworks/default
          subnetworks_self_links.1 = https://www.googleapis.com/compute/beta/projects/REDACTED/regions/us-west2/subnetworks/default
          subnetworks_self_links.10 = https://www.googleapis.com/compute/beta/projects/REDACTED/regions/asia-south1/subnetworks/default
          subnetworks_self_links.11 = https://www.googleapis.com/compute/beta/projects/REDACTED/regions/asia-southeast2/subnetworks/default
          subnetworks_self_links.12 = https://www.googleapis.com/compute/beta/projects/REDACTED/regions/europe-west6/subnetworks/default
          subnetworks_self_links.13 = https://www.googleapis.com/compute/beta/projects/REDACTED/regions/europe-west1/subnetworks/default
          subnetworks_self_links.14 = https://www.googleapis.com/compute/beta/projects/REDACTED/regions/us-central1/subnetworks/default
          subnetworks_self_links.15 = https://www.googleapis.com/compute/beta/projects/REDACTED/regions/us-east1/subnetworks/default
          subnetworks_self_links.16 = https://www.googleapis.com/compute/beta/projects/REDACTED/regions/australia-southeast1/subnetworks/default
          subnetworks_self_links.17 = https://www.googleapis.com/compute/beta/projects/REDACTED/regions/southamerica-east1/subnetworks/default
          subnetworks_self_links.18 = https://www.googleapis.com/compute/beta/projects/REDACTED/regions/europe-west3/subnetworks/default
          subnetworks_self_links.19 = https://www.googleapis.com/compute/beta/projects/REDACTED/regions/asia-southeast1/subnetworks/default
          subnetworks_self_links.2 = https://www.googleapis.com/compute/beta/projects/REDACTED/regions/asia-northeast1/subnetworks/default
          subnetworks_self_links.20 = https://www.googleapis.com/compute/beta/projects/REDACTED/regions/us-west3/subnetworks/default
          subnetworks_self_links.21 = https://www.googleapis.com/compute/beta/projects/REDACTED/regions/asia-northeast2/subnetworks/default
          subnetworks_self_links.22 = https://www.googleapis.com/compute/beta/projects/REDACTED/regions/europe-north1/subnetworks/default
          subnetworks_self_links.23 = https://www.googleapis.com/compute/beta/projects/REDACTED/regions/us-east4/subnetworks/default
          subnetworks_self_links.3 = https://www.googleapis.com/compute/beta/projects/REDACTED/regions/europe-west2/subnetworks/default
          subnetworks_self_links.4 = https://www.googleapis.com/compute/beta/projects/REDACTED/regions/europe-west4/subnetworks/default
          subnetworks_self_links.5 = https://www.googleapis.com/compute/beta/projects/REDACTED/regions/asia-northeast3/subnetworks/default
          subnetworks_self_links.6 = https://www.googleapis.com/compute/beta/projects/REDACTED/regions/us-west4/subnetworks/default
          subnetworks_self_links.7 = https://www.googleapis.com/compute/beta/projects/REDACTED/regions/asia-east2/subnetworks/default
          subnetworks_self_links.8 = https://www.googleapis.com/compute/beta/projects/REDACTED/regions/us-west1/subnetworks/default
          subnetworks_self_links.9 = https://www.googleapis.com/compute/beta/projects/REDACTED/regions/asia-east1/subnetworks/default
        google_compute_instance.target-vm:
          ID = projects/REDACTED/zones/us-central1-a/instances/tf-test-cusom-network-target-vm4kgzu99ytj
          provider = provider.google
          boot_disk.# = 1
          boot_disk.0.auto_delete = true
          boot_disk.0.device_name = persistent-disk-0
          boot_disk.0.disk_encryption_key_raw = 
          boot_disk.0.disk_encryption_key_sha256 = 
          boot_disk.0.initialize_params.# = 1
          boot_disk.0.initialize_params.0.image = https://www.googleapis.com/compute/beta/projects/debian-cloud/global/images/debian-10-buster-v20200805
          boot_disk.0.initialize_params.0.size = 10
          boot_disk.0.initialize_params.0.type = pd-standard
          boot_disk.0.kms_key_self_link = 
          boot_disk.0.mode = READ_WRITE
          boot_disk.0.source = https://www.googleapis.com/compute/v1/projects/REDACTED/zones/us-central1-a/disks/tf-test-cusom-network-target-vm4kgzu99ytj
          can_ip_forward = false
          cpu_platform = Intel Haswell
          current_status = RUNNING
          deletion_protection = false
          description = 
          enable_display = false
          hostname = 
          instance_id = 718961915318796312
          label_fingerprint = 42WmSpB8rSM=
          machine_type = n1-standard-1
          metadata_fingerprint = 2ore-lqyvN0=
          metadata_startup_script = 
          min_cpu_platform = 
          name = tf-test-cusom-network-target-vm4kgzu99ytj
          network_interface.# = 1
          network_interface.0.name = nic0
          network_interface.0.network = https://www.googleapis.com/compute/v1/projects/REDACTED/global/networks/default
          network_interface.0.network_ip = 10.128.0.27
          network_interface.0.subnetwork = https://www.googleapis.com/compute/v1/projects/REDACTED/regions/us-central1/subnetworks/default
          network_interface.0.subnetwork_project = REDACTED
          project = REDACTED
          scheduling.# = 1
          scheduling.0.automatic_restart = true
          scheduling.0.min_node_cpus = 0
          scheduling.0.on_host_maintenance = MIGRATE
          scheduling.0.preemptible = false
          self_link = https://www.googleapis.com/compute/v1/projects/REDACTED/zones/us-central1-a/instances/tf-test-cusom-network-target-vm4kgzu99ytj
          shielded_instance_config.# = 1
          shielded_instance_config.0.enable_integrity_monitoring = true
          shielded_instance_config.0.enable_secure_boot = false
          shielded_instance_config.0.enable_vtpm = true
          tags_fingerprint = 42WmSpB8rSM=
          zone = us-central1-a
        
          Dependencies:
            data.google_compute_image.vmimage
        google_compute_target_instance.custom_network:
          ID = projects/REDACTED/zones/us-central1-a/targetInstances/tf-test-custom-network4kgzu99ytj
          provider = provider.google
          creation_timestamp = 2020-08-10T05:35:16.111-07:00
          description = 
          instance = https://www.googleapis.com/compute/v1/projects/REDACTED/zones/us-central1-a/instances/tf-test-cusom-network-target-vm4kgzu99ytj
          name = tf-test-custom-network4kgzu99ytj
          nat_policy = NO_NAT
          network = 
          project = REDACTED
          self_link = https://www.googleapis.com/compute/v1/projects/REDACTED/zones/us-central1-a/targetInstances/tf-test-custom-network4kgzu99ytj
          zone = https://www.googleapis.com/compute/v1/projects/REDACTED/zones/us-central1-a
        
          Dependencies:
            data.google_compute_network.target-vm
            google_compute_instance.target-vm
        --- FAIL: TestAccComputeTargetInstance_targetInstanceCustomNetworkExample (166.09s)
FAIL
FAIL    github.com/hashicorp/terraform-provider-google-beta/google-beta 168.387s
FAIL
make: *** [testacc] Error 1
````

File: products/compute/terraform.yaml [hunk 1]
```diff
@@ -2430,11 +2430,18 @@ overrides: !ruby/object:Overrides::ResourceOverrides
   TargetInstance: !ruby/object:Overrides::Terraform::ResourceOverride
     examples:
       - !ruby/object:Provider::Terraform::Examples
+        min_version: beta
```
Comment (megan07) [inline products/compute/terraform.yaml:2433]: Oops, I think you added `min_version` to the wrong example :)

Comment (upodroid) [reply products/compute/terraform.yaml]: 😃  fixed

**Comment by megan07** [comment]
@upodroid Thanks for your patience on this! I've created an issue upstream and will merge this after a passing test!

**Comment by megan07** [comment]
@upodroid Thanks again for your patience here, I've followed up with the Google folks and they are getting more information on this issue.

**Comment by megan07** [comment]
Hi @upodroid! I think this may be ready - would you mind rebasing off master and pushing again? Sorry, it's been so long - after that I'll run our suite of tests over it to make sure everything looks good and we can get this merged.
Thanks so much! And sorry again for the delay!

**Comment by upodroid** [comment]
Rebased it to a more recent commit

**Comment by upodroid** [comment]
It is looking good.

```
 REDACTED  MCW0CDP3YY  ~  go  …  github.com  hashicorp  terraform-provider-google-beta   master  29⬇  8✎  8+  USAGE  $    make testacc TEST=./google-beta TESTARGS='-run=TestAccComputeTargetInstance_targetInstanceCustomNetworkExample'
==> Checking source code against gofmt...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 TF_SCHEMA_PANIC_ON_ERROR=1 go test ./google-beta -v -run=TestAccComputeTargetInstance_targetInstanceCustomNetworkExample -timeout 240m -ldflags="-X=github.com/hashicorp/terraform-provider-google-beta/version.ProviderVersion=acc"
=== RUN   TestAccComputeTargetInstance_targetInstanceCustomNetworkExample
=== PAUSE TestAccComputeTargetInstance_targetInstanceCustomNetworkExample
=== CONT  TestAccComputeTargetInstance_targetInstanceCustomNetworkExample
--- PASS: TestAccComputeTargetInstance_targetInstanceCustomNetworkExample (442.58s)
PASS
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta 445.876s
```

**Review by megan07 — APPROVED** [review]
Looks good!  Test passed!

**Pull request merged**